dockstore-webservice/src/main/java/io/dockstore/webservice/resources/OrganizationResource.java

```diff
@@ -443,11 +449,10 @@ public void deleteRejectedOrPendingOrganization(
         @Parameter(description = "Organization ID.", name = "organizationId", in = ParameterIn.PATH, required = true) @PathParam("organizationId") Long organizationId) {
         Organization organization = organizationDAO.findById(organizationId);
         throwExceptionForNullOrganization(organization);
-        OrganizationUser orgUser = getUserOrgRole(organization, user.getId());
+        boolean isUserAdminOrMaintainer = isUserAdminOrMaintainer(organization, user.getId());
 
-        // If the user does not belong to the organization or if the user is not an admin of the organization
-        // and if the user is neither an admin nor curator, then throw an error
-        if ((orgUser == null || orgUser.getRole() != OrganizationUser.Role.ADMIN) && (!user.isCurator() && !user.getIsAdmin())) {
+        // If the user is not an admin or maintainer of the organization and if the user is neither an admin nor curator, then throw an error
+        if (!isUserAdminOrMaintainer && (!user.isCurator() && !user.getIsAdmin())) {
```

Comment:
Should maintainers be able to delete their own org? I would argue that they shouldn't, if they're not allowed to update memberships.

This is pretty edge-casey, as it's unlikely that a deleted or rejected org will have any maintainers to begin with.

Reply:
Will change it back so only the organization admin can delete the org